### PR TITLE
AI-124: Add a feature to provide simple blocking broker request/response ability for components

### PR DESCRIPTION
**Pull request by efunneko**
This PR adds the 'do_broker_request_response' function to ComponentBase to make it very easy for custom components to use broker-attached services.

I have added:
1. A new request_response_flow_controller class to manage an auto-created flow with a single 'broker_request_response' component in it
2. Optional configuration and implementation in ComponentBase to allow a custom component to use this
3. Some tests to ensure it works
4. Documentation on how to use it
5. An example of a custom component making use of it

### CODE REVIEW

File: docs/custom_components.md
Comment (cyrus2281) [inline docs/custom_components.md:1]: There's an example for this in the tips and tricks section, can we add a link to that as well

File: examples/llm/openai_component_request_response.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  - name: main_flow
+    components:
+      # Input from a Solace broker
+      - component_name: input
+        component_module: stdin_input
+
+      # Our custom component
+      - component_name: llm_streaming_custom_component
+        component_module: llm_streaming_custom_component
+        component_base_path: examples/llm/custom_components
```
Comment (cyrus2281) [inline examples/llm/openai_component_request_response.yaml:68]: Maybe add a comment saying this is relative path

File: src/solace_ai_connector/solace_ai_connector.py [hunk 2]
```diff
@@ -98,15 +100,6 @@ def wait_for_flows(self):
...
 
-    def stop(self):
-        """Stop the Solace AI Event Connector"""
-        log.info("Stopping Solace AI Event Connector")
-        self.stop_signal.set()
-        self.timer_manager.stop()  # Stop the timer manager first
-        self.wait_for_flows()
-        if self.trace_thread:
-            self.trace_thread.join()
-
```
Comment (cyrus2281) [inline src/solace_ai_connector/solace_ai_connector.py:109]: Why was the stop function remove? It's also being reference in few places, eg line 100

File: src/solace_ai_connector/components/inputs_outputs/stdin_input.py [hunk 3]
```diff
@@ -9,7 +11,9 @@
     "class_name": "Stdin",
     "description": (
         "STDIN input component. The component will prompt for "
-        "input, which will then be placed in the message payload using the output schema below."
+        "input, which will then be placed in the message payload using the output schema below. "
+        "The component will wait for its output message to be acknowledged before prompting for "
+        "the next input."
     ),
     "config_parameters": [],
```
Comment (cyrus2281) [inline src/solace_ai_connector/components/inputs_outputs/stdin_input.py:18]: "prompt" is not listed under the possible config parameters

File: examples/llm/openai_component_request_response.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
+# This example demostrates how to use the request_response_flow_controller to 
+# inject another flow into an existing flow. This is commonly used when
+# you want to call a service that is only accessible via the broker.
+#
+#    Main flow:   STDIN -> llm_streaming_custom_component -> STDOUT
+#                                          |
+#                                          v
+#                              do_broker_request_response()
```
Comment (cyrus2281) [inline examples/llm/openai_component_request_response.yaml:8]: Doesn't this wait for the response before going to STDOUT? The graphs convey it branches into 2 different flows. Maybe add a return arrow from `do_broker_request_response`

**Review by cyrus2281 — CHANGES_REQUESTED** [review]
Some concerns about solace_ai_connector.py stop function + some small comments

File: docs/custom_components.md
Comment (efunneko) [reply docs/custom_components.md:1]: Added

File: examples/llm/openai_component_request_response.yaml (hunk 4 above)
Comment (efunneko) [reply examples/llm/openai_component_request_response.yaml:8]: Good point. fixed

**Review by cyrus2281 — APPROVED** [review]

**Pull request merged**